src/gemdat/transitions.py

```diff
@@ -405,3 +406,197 @@ def _calculate_occupancy(atom_sites: np.ndarray) -> dict[int, int]:
     occupancy = dict(zip(unq, counts))
     occupancy.pop(NOSITE, None)
     return occupancy
+
+
+def _wrap_pbc(coord: np.ndarray, size: tuple) -> tuple:
+    """Wraps coordinates in periodic boundaries.
+
+    Parameters
+    ----------
+    coord: np.ndarray
+        Coordinates of one or multiple sites
+    size: tuple
+        Boudnary size
+
+    Returns:
+    -------
+    wrapped_coord: tuple
+        Coordinates inside the PBC
+    """
+    wrapped_coord = tuple(coord % size)
+    return wrapped_coord
+
+
+def free_energy_graph(F: np.ndarray,
+                      max_energy_threshold=1e20,
+                      diagonal=True) -> nx.Graph:
+    """Compute the graph of the free energy for networkx functions.
+
+    Parameters
+    ----------
+    F : np.ndarray
+        Free energy on the 3d grid
+    max_energy_threshold : float, optional
+        Maximum energy threshold for the path to be considered valid
+    diagonal : bool
+        If True, allows diagonal grid moves
+
+    Returns
+    -------
+    G : nx.Graph
+        Graph of free energy
+    """
+
+    # Define possible movements in 3D space
+    if not diagonal:
+        movements = [(1, 0, 0), (-1, 0, 0), (0, 1, 0), (0, -1, 0), (0, 0, 1),
+                     (0, 0, -1)]
+    else:
+        movements = [(1, 0, 0), (-1, 0, 0), (0, 1, 0), (0, -1, 0), (0, 0, 1),
+                     (0, 0, -1),
+                     (1, 1, 0), (-1, -1, 0), (1, -1, 0), (-1, 1, 0), (1, 0, 1),
+                     (-1, 0, -1), (1, 0, -1), (-1, 0, 1), (0, 1, 1),
+                     (0, -1, -1), (0, 1, -1), (0, -1, 1), (1, 1, 1),
+                     (-1, -1, -1), (1, -1, -1), (-1, 1, 1)]
+
+    G = nx.Graph()
+    for i in range(F.shape[0]):
+        for j in range(F.shape[1]):
+            for k in range(F.shape[2]):
+                if F[i, j, k] >= 0 and F[i, j, k] < max_energy_threshold:
+                    G.add_node((i, j, k), energy=F[i, j, k])
```

Comment:
Use ndenumerate: https://numpy.org/doc/stable/reference/generated/numpy.ndenumerate.html

```suggestion

    G = nx.Graph()
    for index, val in np.ndenumerate(F):
        if 0 <= energy < max_energy_threshold:
             G.add_node(index, energy=energy)
```

Reply:
nice, thanks